metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,424 @@
+    except MetaflowNotFound:
+        raise CommandException("Flow *%s* does not exist " % flow_name +
+                               "in any namespace.\n" +
+                               "Please run the flow before tagging.")
+    except MetaflowNamespaceMismatch:
+        raise CommandException("Flow *%s* does not belong to " % flow_name +
+                               "namespace *%s*.\n" % user_namespace +
+                               "Check typos if you used option --namespace.")
+
+    # throw an error with message to include latest run-id when run_id is None
+    if run_id is None:
+        latest_run_id = Flow(pathspec=flow_name).latest_run.id
+        msg = "Please specify a run-id using --run-id\n" + \
+              "Flow %s's latest run has run-id *%s* under namespace %s" \
+            % (flow_name, latest_run_id, user_namespace)
+        raise CommandException(msg)
+    run_id_parts = run_id.split('/')
+    if len(run_id_parts) == 1:
+        run_name = "%s/%s" % (flow_name, run_id)
+    elif len(run_id_parts) == 2:
+        if run_id_parts[0] != flow_name:
+            raise CommandException(
+                "Requesting run-id *%s* which is not a run of flow %s" % (run_id, flow_name))
+        run_name = run_id
+    else:
+        raise CommandException("Run-id *%s* is not a valid run-id" % run_id)
+
+    # handle error messaging for three cases
+    # 1. our user makes a typo in --run-id
+    # 2. our user tries to mutate tags on a meson/production run
+    # 3. our user's --run-id does not exist in the default/specified namespace
+    try:
+        namespace(user_namespace)
+        metaflow_client = Run(pathspec=run_name)
+    except MetaflowNotFound:
+        raise CommandException("Run *%s* does not exist " % run_name +
+                               "in any namespace.\nPlease check your --run-id.")
+    except MetaflowNamespaceMismatch:
+        namespace(None)
+        metaflow_client = Run(pathspec=run_name)
+
+        run_tags = metaflow_client.tags
+
+        if any([is_prod_token(tag) for tag in run_tags]):
+            raise CommandException(
+                "Modifying a scheduled run's tags is currently not allowed.")
+
+        msg = "Run *%s* does not belong to " % run_name + \
+            "namespace *%s*.\n" % user_namespace
+
+        user_tags = [tag for tag in run_tags if tag.startswith("user:")]
+        if user_tags:
+            msg += "You can choose any of the following tags associated with *%s* " % run_name + \
+                   "to use with option *--namespace*\n"
+            msg += "\n".join(["\t*%s*" % tag for tag in user_tags])
+
+        raise CommandException(msg)
+
+    return metaflow_client
+
+
+def list_runs(obj, user_namespace):
+    namespace(user_namespace)
+    metaflow_client = Flow(pathspec=obj.flow.name)
+    return metaflow_client.runs()
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help='Commands related to Metaflow tagging.')
+def tag():
```

Comment:
Minor nit - Should we call it `tags` rather than `tag`? Okay, either way.

Reply:
Not changing to avoid breaking compatibility with internal commands.